### PR TITLE
enh(wizard): Register informations from Wizard into platform_topology table

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR allow the Wizard to register informations into platform_topology table. 
Also, the Select Remote Server autocompletion is now made from platform_topology, and an autocompletion for Poller part has been added.

**Fixes** # MON-5866 MON-5867

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.04.x
- [ ] 19.10.x
- [ ] 20.04.x
- [x] 20.10.x (master)

<h2> How this pull request can be tested ? </h2>

- Create a new poller  / Remote Server into the wizard, you should see it into the platform_topology table.
- Create a new poller / Remote Server using the Radio button "Select a Poller / Remote Server", you should select some pending Remote / Poller.

## Checklist

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: src/CentreonRemote/Application/Webservice/CentreonConfigurationRemote.php [hunk 1]
```diff
@@ -422,15 +453,36 @@ public function postLinkCentreonRemoteServer(): array
                 $noProxy
             );
             $this->setCentreonInstanceAsCentral();
-
-        // if it is poller wizard and poller is linked to another poller/remote server (instead of central)
+            $this->updateServerInPlatformTopology([
+                'type' => 'remote',
```
Comment (kduret) [inline src/CentreonRemote/Application/Webservice/CentreonConfigurationRemote.php:457]: use `PlatformTopology::TYPE_REMOTE`

File: src/CentreonRemote/Application/Webservice/CentreonConfigurationRemote.php [hunk 2]
```diff
@@ -422,15 +453,36 @@ public function postLinkCentreonRemoteServer(): array
...
+            // if it is poller wizard and poller is linked to another poller/remote server (instead of central)
         } elseif ($pollerConfigurationBridge->hasPollersForUpdating()) {
             $pollers = [$pollerConfigurationBridge->getPollerFromId($serverId)];
             $parentPoller = $pollerConfigurationBridge->getLinkedPollersSelectedForUpdate()[0];
             $pollerConfigurationService->linkPollersToParentPoller($pollers, $parentPoller);
             // add broker output to forward data to additionnal remote server and link in db
             $additionalRemotes = $pollerConfigurationBridge->getAdditionalRemoteServers();
             $pollerConfigurationService->linkPollerToAdditionalRemoteServers($pollers[0], $additionalRemotes);
+            $this->updateServerInPlatformTopology([
+                'type' => 'poller',
```
Comment (kduret) [inline src/CentreonRemote/Application/Webservice/CentreonConfigurationRemote.php:473]: use `PlatformTopology::TYPE_POLLER`

File: src/CentreonRemote/Application/Webservice/CentreonConfigurationRemote.php [hunk 3]
```diff
@@ -422,15 +453,36 @@ public function postLinkCentreonRemoteServer(): array
...
+                'server_name' => $serverName,
+                'nagios_id' => $serverId,
+                'address' => $serverIP,
+                'parent' => $parentPoller->getId()
+            ]);
+        } else {
+            $this->updateServerInPlatformTopology([
+                'type' => 'poller',
```
Comment (kduret) [inline src/CentreonRemote/Application/Webservice/CentreonConfigurationRemote.php:481]: use `PlatformTopology::TYPE_POLLER`

File: www/front_src/src/components/forms/poller/PollerFormStepOne.js [hunk 4]
```diff
@@ -9,13 +9,51 @@ import { withTranslation } from 'react-i18next';
...
+      inputTypeManual: !value,
+    });
+  };
+
+  UNSAFE_componentWillReceiveProps = (nextProps) => {
+    const { waitList } = nextProps;
+    const { initialized } = this.state;
+    if (waitList && !initialized) {
+      this.initializeFromRest(waitList.length > 0);
+      // this.initializeFromRest(true);//set to true of false if abandon the upper case condition
```
Comment (kduret) [inline www/front_src/src/components/forms/poller/PollerFormStepOne.js:41]: remove comments

**Review by kduret — DISMISSED** [review]

File: src/CentreonRemote/Application/Webservice/CentreonConfigurationRemote.php [hunk 5]
```diff
@@ -2,9 +2,11 @@
 
 namespace CentreonRemote\Application\Webservice;
 
-use CentreonRemote\Application\Validator\WizardConfigurationRequestValidator;
+use Exception;
```
Comment (callapa) [inline src/CentreonRemote/Application/Webservice/CentreonConfigurationRemote.php:5]: No need to do this, use this class like this: `\Exception.`

File: src/CentreonRemote/Application/Webservice/CentreonConfigurationRemote.php [hunk 6]
```diff
@@ -327,11 +347,23 @@ public function postLinkCentreonRemoteServer(): array
...
+        ) {
             return ['error' => true, 'message' => "Invalid IP address"];
         }
-
         $dbAdapter = $this->getDi()['centreon.db-manager']->getAdapter('configuration_db');
+
+        /**
+         * Avoid Ip duplication
+         */
+        $dbAdapter->query('SELECT * FROM `nagios_server` WHERE `ns_ip_address` = ?', [$serverIP]);
```
Comment (callapa) [inline src/CentreonRemote/Application/Webservice/CentreonConfigurationRemote.php:360]: Use SELECT COUNT(*) instead.

**Review by callapa — DISMISSED** [review]

File: www/front_src/src/components/forms/remoteServer/RemoteServerFormStepOne.js [hunk 7]
```diff
@@ -43,14 +43,18 @@ class RemoteServerFormStepOne extends Component {
...
     }
     this.setState({
       initialized: true,
       centreon_folder: '/centreon/',
     });
   };
 
+  handleChange = (e, value) => {
+    const platform = this.props.waitList.find((server) => server.ip === value);
+    this.props.change('server_name', platform.server_name);
```
Comment (Thebarda) [inline www/front_src/src/components/forms/remoteServer/RemoteServerFormStepOne.js:55]:
```suggestion
   const { waitList, change } = this.props;
    const platform = waitList.find((server) => server.ip === value);
    change('server_name', platform.server_name);
```

**Review by bdauria — DISMISSED** [review]

File: src/CentreonRemote/Application/Webservice/CentreonConfigurationRemote.php [hunk 8]
```diff
@@ -327,11 +346,27 @@ public function postLinkCentreonRemoteServer(): array
...
+         * Avoid Ip duplication
+         */
+        $statement = $this->pearDB->prepare(
+            'SELECT COUNT(*) as `total` FROM `nagios_server` WHERE `ns_ip_address` = :serverIp'
+        );
+        $statement->bindValue(':serverIp', $serverIP, \PDO::PARAM_STR);
+        $statement->execute();
+        $isInNagios = $statement->fetch(\PDO::FETCH_ASSOC);
+        if ((int) $isInNagios['total'] > 0) {
+            throw new \Exception('This IP Address already exist');
```
Comment (sc979) [inline src/CentreonRemote/Application/Webservice/CentreonConfigurationRemote.php:367]:
```suggestion
            throw new \Exception(_('This IP Address already exist'));
```

**Review by Thebarda — APPROVED** [review]

**Review by sc979 — APPROVED** [review]

**Pull request merged**